libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1887,20 +1964,39 @@ string YulUtilFunctions::negateNumberCheckedFunction(Type const& _type)
 string YulUtilFunctions::zeroValueFunction(Type const& _type)
 {
 	solUnimplementedAssert(_type.sizeOnStack() == 1, "Stacksize not yet implemented!");
-	solUnimplementedAssert(_type.isValueType(), "Zero value for non-value types not yet implemented");
+	solAssert(_type.category() != Type::Category::Mapping, "");
 
 	string const functionName = "zero_value_for_" + _type.identifier();
 
+	Whiskers templ(R"(
```

Comment:
It is better to move all this inside the lambda function because then it will only be executed once per type.

Reply:
Done